File: AspNetMvcUnleashed/0672329980_source/Chapter17/CS/UnleashedBlog/UnleashedBlog.Tests/Controllers/BlogControllerTests.cs
using System.Collections;
using System.Web.Mvc;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using UnleashedBlog.Controllers;
using UnleashedBlog.Models;
using UnleashedBlog.Tests.Models;

namespace UnleashedBlog.Tests.Controllers
{
    [TestClass]
    public class BlogControllerTests
    {

        [TestMethod]
        public void ShowNewBlogEntries()
        {
            // Arrange
            var repository = new FakeBlogRepository();
            var controller = new BlogController(repository);

            // Act
            var result = (ViewResult)controller.Index();

            // Assert
            CollectionAssert.AllItemsAreInstancesOfType((ICollection)result.ViewData.Model, typeof(BlogEntry));
        }


        [TestMethod]
        public void CreateBlogEntry()
        {
            // Arrange
            var repository = new FakeBlogRepository();
            var controller = new BlogController(repository);
            var blogEntryToCreate = new BlogEntry();

            // Act
            controller.Create(blogEntryToCreate);
            var result = (ViewResult)controller.Index();

            // Assert
            var firstEntry = ((IList)result.ViewData.Model)[0];
            Assert.AreSame(blogEntryToCreate, firstEntry);
        }



    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter17/VB/UnleashedBlog/UnleashedBlog.Tests/Controllers/BlogControllerTests.vb
Imports System
Imports System.Text
Imports System.Collections.Generic
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Web.Mvc

<TestClass()> Public Class BlogControllerTests

    <TestMethod()> _
    Public Sub ShowNewBlogEntries()
        ' Arrange
        Dim repository As New FakeBlogRepository()
        Dim controller = New BlogController(repository)

        ' Act
        Dim result As ViewResult = controller.Index()

        ' Assert
        CollectionAssert.AllItemsAreInstancesOfType(result.ViewData.Model, GetType(BlogEntry))
    End Sub


    <TestMethod()> _
    Public Sub CreateBlogEntry()
        ' Arrange
        Dim repository As New FakeBlogRepository()
        Dim controller = New BlogController(repository)
        Dim blogEntryToCreate = New BlogEntry()

        ' Act
        controller.Create(blogEntryToCreate)
        Dim result As ViewResult = controller.Index()

        ' Assert
        Dim firstEntry = result.ViewData.Model(0)
        Assert.AreSame(blogEntryToCreate, firstEntry)
    End Sub

End Class
